modules/drivers/sqlite/src/metabase/driver/sqlite.clj

```diff
@@ -281,8 +291,11 @@
 
 (defmethod sql.qp/->honeysql [:sqlite :ceil]
   [_driver [_ arg]]
-  (hsql/call :round (hsql/call :+ arg 0.5)))
-
+  (hsql/call :case
+    ;; check if we're ceiling a whole number, it should return itself
+    ;; [:ceil 1.0] should returns 1
+    (hsql/call := (hsql/call :round arg) arg) arg
```

Comment:
wouldn't this return `1.0` since it is returning arg? Do you need to return the `(hsql/call :round arg)` or truncate the number to an integer?

Reply:
ah right, I need to cast it to an integer. Thanks!